src/Adapter/ContextLoadHelper.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter;
+
+use Context;
+use Currency;
+use DummyController;
+use Employee;
+use Shop;
+
+/**
+ * Helps loading specific context, for example in CLI context
+ */
+class ContextLoadHelper
+{
+    /**
+     * @var Context
+     */
+    private $context;
+
+    /**
+     * @param LegacyContext $context
+     */
+    public function __construct(LegacyContext $context)
+    {
+        $this->context = $context->getContext();
+    }
+
+    /**
+     * @param string|null $controllerClassName
+     * @param int|null $currencyId
+     * @param int|null $employeeId
+     * @param int|null $shopId
+     * @param int|null $shopGroupId
+     */
+    public function loadGenericContext(
+        ?string $controllerClassName = null,
+        ?int $currencyId = null,
+        ?int $employeeId = null,
+        ?int $shopId = null,
+        ?int $shopGroupId = null
+    ) {
+        $this->loadCurrencyContext($currencyId);
+        $this->loadControllerContext($controllerClassName);
+        $this->loadEmployeeContext($employeeId);
+
+        if (null !== $shopId) {
+            $this->loadShopContext($shopId);
+        }
+        if (null !== $shopGroupId) {
+            $this->loadShopGroupId($shopGroupId);
+        }
+    }
+
+    /**
+     * @param string|null $controllerClassName
+     */
+    public function loadControllerContext(?string $controllerClassName = null)
```

Comment:
You should return `$this` in all these load functions, this would allow to chain the calls:

```php
$this
    ->loadCurrencyContext($currencyId)
    ->loadControllerContext($controllerClassName)
    ->loadEmployeeContext($employeeId)
;
```

Reply:
Fixed in https://github.com/PrestaShop/PrestaShop/pull/21125/commits/2b89cac34e6ebcf56b999f38faf4277a7b485d0f